cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -88,23 +88,23 @@ fn matches_input_any<C: LowerCtx<I = Inst>>(
 
 /// Emits instruction(s) to generate the given 64-bit constant value into a newly-allocated
 /// temporary register, returning that register.
-fn generate_constant<C: LowerCtx<I = Inst>>(ctx: &mut C, ty: Type, c: u64) -> Reg {
+fn generate_constant<C: LowerCtx<I = Inst>>(ctx: &mut C, ty: Type, c: u64) -> ValueRegs<Reg> {
     let from_bits = ty_bits(ty);
     let masked = if from_bits < 64 {
         c & ((1u64 << from_bits) - 1)
     } else {
         c
     };
 
-    let cst_copy = ctx.alloc_tmp(Inst::rc_for_type(ty).unwrap(), ty);
-    for inst in Inst::gen_constant(cst_copy, masked, ty, |reg_class, ty| {
-        ctx.alloc_tmp(reg_class, ty)
+    let cst_copy = ctx.alloc_tmp(ty);
+    for inst in Inst::gen_constant(cst_copy, masked as u128, ty, |_reg_class, ty| {
```

Comment:
(Second time I see an instance of this, so asking, but it might be false.) Could it be that the `reg_class` arg is now always ignored, since the type is used instead? If so, we could change the signature of `gen_constant` and all its callers to reflect this.

Reply:
Yes, good point; I went ahead and carried this refactor through all the backends.